rmw_fastrtps_shared_cpp/test/test_rmw_qos_to_dds_attributes.cpp

```diff
@@ -94,7 +95,27 @@ TEST_F(GetDataReaderQoSTest, nominal_conversion) {
   EXPECT_EQ(
     eprosima::fastrtps::KEEP_LAST_HISTORY_QOS,
     subscriber_attributes_.topic.historyQos.kind);
-  EXPECT_EQ(10, subscriber_attributes_.topic.historyQos.depth);
+  EXPECT_GE(10, subscriber_attributes_.topic.historyQos.depth);
+}
+
+TEST_F(GetDataReaderQoSTest, large_depth_conversion) {
+  size_t depth = subscriber_attributes_.topic.historyQos.depth + 1;
```

Comment:
What is the general type of depth here? Are you relying on integer overflowing?

Reply:
Specifically, "signed integer overflowing"

Reply:
Type is `size_t` for `rmw` and `int32_t` for `fastrtps`. This isn't relying on overflow (which are UB in C, IIRC), simply ensuring the configured `depth` is larger than the one already configured. 

I don't think there's a practical risk of overflow here anyways, unless we start running ROS 2 on 16 bits architectures or eProsima starts using `std::numeric_limits<int32_t>::max()` as default depth.

Reply:
Ok, I think I understand better what's being tested.